index.ts

```diff
@@ -113,40 +152,35 @@ function candidateToGeojson (candidate) {
  * @param {string} [$0.url]                       optional URL to override ESRI suggest endpoint
  * @return {Promise}                              A Promise that'll get resolved with the suggest result
  */
-export function autocomplete ({
+export function autocomplete({
+  boundary,
   clientId,
   clientSecret,
-  boundary,
   focusPoint,
   text,
   url
-}: BaseQuery & {
-  boundary?: Boundary,
-  focusPoint?: any,
-  text: string,
-}): Promise<BaseResponse> {
-  const geocoder = getGeocoder(clientId, clientSecret, url)
-  const options = {}
+}: BaseQuery & Options): Promise<BaseResponse> {
```

Comment:
Two things:

1. Can you in-line the applicable keys within the `Options` type since not all of those values would apply as keys in the input argument for this function call?
2. Can you also please extend the `BaseResponse` type with the `query` key/value since the query is eventually returned?